physics/SFC_Layer/UFS/sfc_diff.f

```diff
@@ -425,17 +437,6 @@ subroutine sfc_diff_run (im,rvrdm1,eps,epsm1,grav,                &  !intent(in)
                  z0rl_wat(i) = 1.0e-4_kp
               endif
 
-            elseif (z0rl_wav(i) <= 1.0e-7_kp .or.                       &
```

Comment:
@JongilHan66 I'm concerned that removal of this section of code is having some unintended effects.  It's my understanding is that this code was put in by Moorthi after we noticed there were issues with where the atm model expected z0 from waves, and the vlaues that the wave model sent, whether it was land/sea masks or a value the atm model did not thinkk was acceptable.  I ran a few of the regression tests offline and I saw this: 


 
![showdiffcpldbmktest](https://github.com/ufs-community/ccpp-physics/assets/22549231/7d31f7df-55b8-4e28-b16f-e2d898833f92)


On the left is the differences and on the top right is from this branch and the bottom right is from develop.  It's easy to see this issue by the differences being near land/sea masks and in lakes.     Please let me know if you'd like to discuss this more offline and sorry for the delay in looking closer at this before.

Reply:
@JessicaMeixner-NOAA I thought that section would not be necessary as molecular viscosity effect is now included (which will result in z0 > 1.e-7 always). But I have no objection to recover that section for a safety.

Reply:
@JongilHan66 I think focusing on for example the lakes really shows where this is needed.  Otherwise it defaults to the minimum value because the waves do not provide a roughness length there.

Reply:
@Qingfu-Liu @JessicaMeixner-NOAA Hi Qingfu, could you recover that section to the original one, i.e.,

!           elseif (z0rl_wav(i) <= 1.0e-7_kp .or.                       &
!    &              z0rl_wav(i) > 1.0_kp) then
!!            z0 = (charnock / grav) * ustar_wat(i) * ustar_wat(i)
!             tem1 = 0.11 * vis / ustar_wat(i)
!             z0 = tem1 + (charnock/grav)*ustar_wat(i)*ustar_wat(i)

!             if (redrag) then
!               z0rl_wat(i) = 100.0_kp * max(min(z0, z0s_max),1.0e-7_kp)
!             else
!               z0rl_wat(i) = 100.0_kp * max(min(z0,0.1_kp), 1.0e-7_kp)
!             endif
!           endif

==>
            elseif (z0rl_wav(i) <= 1.0e-7_kp .or.                       &
     &              z0rl_wav(i) > 1.0_kp) then
!             z0 = (charnock / grav) * ustar_wat(i) * ustar_wat(i)
              tem1 = 0.11 * vis / ustar_wat(i)
              z0 = tem1 + (charnock/grav)*ustar_wat(i)*ustar_wat(i)

              if (redrag) then
                z0rl_wat(i) = 100.0_kp * max(min(z0, z0s_max),1.0e-7_kp)
              else
                z0rl_wat(i) = 100.0_kp * max(min(z0,0.1_kp), 1.0e-7_kp)
              endif
            endif

Reply:
@JongilHan66  I have added this section of original code and tested in suite control_p8. Can you review the new code? Thanks

Reply:
@Qingfu-Liu I saw that that section was correctly recovered to the original one.